### DISCUSSION TITLE
is the order important?

### DISCUSSION THREAD

**Discussion by jdvsch** [react-hook-form/react-hook-form#9651] [Q&A] [answered]
Hello everyone,
I appreciate if you help me solving a doubt... Usually I see that the schema is created after the imports... Is there a problem if I create it as I show in the following example?

`import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from 'yup';

const App = () => {

const schema = yup.object().shape({
    name: yup.string().required()
  }) .required();

  const { register, handleSubmit } = useForm({
    resolver: yupResolver(schema),
  });

  return (
    <form onSubmit={handleSubmit((d) => console.log(d))}>
      <input {...register('name')} />
      <input type="submit" />
    </form>
  );
};`

**Comment by Moshyfawn** [comment, marked as answer]
It's fine to define your schema inside the func component body. One thing you can't do is infer a type from your schema. But if you don't need to do that, it's fine.

Keep in mind that changing the schema shape isn't reactive: it won't be reflected in your resolver logic

**Reply by jdvsch** [reply]
Thanks for your help and quick response Moshyfawn
